files/conf/k8s-install-binaries

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+set -euo pipefail
+
+image=$1
+version=$2
+
+semver=( ${version//./ } )
+minor="${semver[1]}"
+
+declare -a binaries=(kubectl kubelet)
+
+docker create --name hyperkube "$image"
+if [ "$minor" -lt "17" ]; then
+  docker cp hyperkube:/hyperkube /opt/bin/hyperkube
+fi
+for binary in "${binaries[@]}"; do
+  if [ "$minor" -lt "17" ]; then
+    echo "hyperkube $binary "'"$@"' > /opt/bin/"$binary"
```

Comment:
I don't understand how this works / what is happening here. Can you explain it to me?

Reply:
Sure. This creates two wrapper scripts, `/opt/bin/kubectl` and `/opt/bin/kubelet`. They have a single line, `hyperkube kubectl "$@"` and `hyperkube kubelet "$@"` respectively, which just redirects all arguments to `hyperkube` setting the first argument to the name of the `hyperkube` bundled binary. `kubelet --arg1` then becomes `hyperkube kubelet --arg1`.

Reply:
Okay - that makes sense to me. Can you maybe add a comment explaining this? It might be only me but this is some serious bash magic to me.

Reply:
Good call--I'll try to spin up a cluster with this, add some comments, then open it up for more review.